### ISSUE TITLE
Incorrect unreachability inference with nested try blocks and a forever loop

### ISSUE THREAD

**Issue by jonathandung** [astral-sh/ty#3512]
### Summary

```python
from secrets import randbelow
from time import sleep
from typing import assert_never
x = 0
try:
    while True:
        try:
            print('spam')
            if not (x := randbelow(3)): 1/0
        except KeyboardInterrupt: sleep(1)
finally: assert_never(x) # This fails at runtime because it is run when `randbelow` returns 0 (with a 1 in 3 chance)
# but ty thinks the branch is indeed unreachable and produces no diagnostic
```

### Version

ty 0.0.38 (1d3efc1d6 2026-05-20)

**Comment by AlexWaygood** [comment]
Thanks! I think this is #233
